docs/01-app/03-api-reference/03-file-conventions/01-metadata/opengraph-image.mdx

```diff
@@ -220,25 +220,33 @@ The default export function receives the following props:
 
 #### `params` (optional)
 
-An object containing the [dynamic route parameters](/docs/app/api-reference/file-conventions/dynamic-routes) object from the root segment down to the segment `opengraph-image` or `twitter-image` is colocated in.
+A promise that resolves to an object containing the [dynamic route parameters](/docs/app/api-reference/file-conventions/dynamic-routes) object from the root segment down to the segment `opengraph-image` or `twitter-image` is colocated in.
+
+> **Good to know**: If you use [`generateImageMetadata`](/docs/app/api-reference/functions/generate-image-metadata), the function will also receive an `id` prop that is a promise resolving to the `id` value from one of the items returned by `generateImageMetadata`.
 
 ```tsx filename="app/shop/[slug]/opengraph-image.tsx" switcher
-export default function Image({ params }: { params: { slug: string } }) {
+export default async function Image({
+  params,
+}: {
+  params: Promise<{ slug: string }>
+}) {
+  const { slug } = await params
   // ...
 }
 ```
 
 ```jsx filename="app/shop/[slug]/opengraph-image.js" switcher
-export default function Image({ params }) {
+export default async function Image({ params }) {
+  const { slug } = await params
   // ...
 }
 ```
 
-| Route                                      | URL         | `params`                  |
-| ------------------------------------------ | ----------- | ------------------------- |
-| `app/shop/opengraph-image.js`              | `/shop`     | `undefined`               |
-| `app/shop/[slug]/opengraph-image.js`       | `/shop/1`   | `{ slug: '1' }`           |
-| `app/shop/[tag]/[item]/opengraph-image.js` | `/shop/1/2` | `{ tag: '1', item: '2' }` |
+| Route                                      | URL         | `params`                           |
+| ------------------------------------------ | ----------- | ---------------------------------- |
+| `app/shop/opengraph-image.js`              | `/shop`     | `Promise<undefined>`               |
```

Comment:
@huozhi I guess this one is technically correct though? or will the entire params object be `undefined`?